### PR TITLE
COG-89 Administrators own account is Read Only

### PR DESCRIPTION
**Pull request by Nasir2712**
make the admin account ready only by disabling the edit button on user detail.

### CODE REVIEW

File: lib/perry/account.rb [hunk 1]
```diff
@@ -19,5 +19,6 @@ class Account < Dry::Struct
     attribute :county_name, Types::String.optional
     attribute :roles, Types::Array.of(Types::String).optional
     attribute :privileges, Types::Array.of(Types::String).optional
+    attribute :userName, Types::String.optional
```
Comment (RamuKammagani-osi) [inline lib/perry/account.rb]: Are you sure that it is userName?? i guess it might be user_name.

Comment (Nasir2712) [reply lib/perry/account.rb]: ya, it is userName.

Comment (Nasir2712) [reply lib/perry/account.rb]: yup this is what they are using.

File: app/javascript/selectors/detailSelector.js [hunk 2]
```diff
@@ -11,3 +11,25 @@ export const selectDetailRecords = state => {
 
 export const permissionsList = state =>
   safeGet(state, 'fetchPermissions.permissions.permissions', []);
+
+const getAccountUserName = state =>
+  state.fetchAccount.account && state.fetchAccount.account.account.userName
+    ? state.fetchAccount.account.account.userName
+    : '';
+const getUserDetailRecordId = state => {
```
Comment (RamuKammagani-osi) [inline app/javascript/selectors/detailSelector.js]: shouldn't this be "getUserDetailRecords" instead of "getUserDetailRecordId" as this returns records!

Comment (Nasir2712) [reply app/javascript/selectors/detailSelector.js]: it returns userDetailRecordId, check line 26.

File: app/javascript/selectors/detailSelector.js [hunk 3]
```diff
@@ -11,3 +11,25 @@ export const selectDetailRecords = state => {
...
+  const userDetailRecord =
+    state.fetchDetails.details !== null
+      ? state.fetchDetails.details.records
+      : '';
```
Comment (kelfink) [inline app/javascript/selectors/detailSelector.js]: records is an array or object, but '' is the default if details is null?  Should it be an {} or [] maybe ?

File: app/javascript/selectors/detailSelector.test.js [hunk 4]
```diff
@@ -100,4 +104,137 @@ describe('selectors', () => {
...
+          fetchDetails: {
+            details: null,
+            fetching: false,
+            error: null,
+          },
+          fetchAccount: {
+            account: {
+              XHRStatus: 'ready',
+              account: {
+                userName: null,
```
Comment (kelfink) [inline app/javascript/selectors/detailSelector.test.js]: This state doesn't agree with the it block description.  only details null?  account username is null as well, which I think is the sam was the case below.

Comment (kelfink) [reply app/javascript/selectors/detailSelector.test.js]: Not sure we need both cases.

File: app/javascript/selectors/detailSelector.js [hunk 5]
```diff
@@ -11,3 +11,25 @@ export const selectDetailRecords = state => {
...
+  const userDetailRecordId =
+    userDetailRecord !== '' ? state.fetchDetails.details.records.id : '';
+  return userDetailRecordId;
+};
+
+export const checkEditDisableBtn = state => {
+  const accountUserName = getAccountUserName(state);
+  const userDetailRecordId = getUserDetailRecordId(state);
+  if (!(accountUserName === '' || userDetailRecordId === '')) {
```
Comment (kelfink) [inline app/javascript/selectors/detailSelector.js]: I'd prefer ```if (accountUserName !== ''  && userDetailRecordId !== '') {```.  This eliminates the negating wrapper and reads more clearly.

Comment (kelfink) [reply app/javascript/selectors/detailSelector.js]: Also, the methods being called are using '' to represent null values.  Can they return '' when the originating value not null?  Maybe it doesn't matter

File: app/javascript/containers/detailsContainer.js [hunk 6]
```diff
@@ -3,6 +3,7 @@ import UserDetail from '../views/userDetail/UserDetail';
 import {
   selectDetailRecords,
   permissionsList,
+  checkEditDisableBtn,
```
Comment (ratneshraval) [inline app/javascript/containers/detailsContainer.js]: I would like it it said `checkEditDisabledBtn`. More readable.

File: app/javascript/selectors/detailSelector.js (hunk 5 above)
Comment (ratneshraval) [reply app/javascript/selectors/detailSelector.js]: Once you set else to null you can simply do ` accountUserName && userDetailRecordId `

File: app/javascript/selectors/detailSelector.test.js (hunk 4 above)
Comment (Nasir2712) [reply app/javascript/selectors/detailSelector.test.js]: removed a case where username was null and details were null.

**Comment by Nasir2712** [comment]
@kelfink I am not sure if the API is prohibiting this behavior in Perry. I think @sorych can answer your question.

**Pull request merged**